[NFC][Flang] Move bounds helper functions to Util header.

TIFitis opened the pull request:
This PR moves the `needsBoundsOps` and `genBoundsOps` helper functions to `flang/include/flang/Optimizer/OpenMP/Utils.h`.

bhandarkar-pranav reviewed (approved):
Thanks @TIFitis - LGTM. Just one super micro nit.

TIFitis committed:
[NFC][Flang] Move bounds helper functions to Util header.
This PR moves the needsBoundsOps and genBoundsOps helper functions to flang/include/flang/Optimizer/OpenMP/Utils.h.
